protocol/x/vault/keeper/orders_test.go

```diff
@@ -1 +1,424 @@
+			ExchangeConfigJson: constants.TestMarketParams[1].ExchangeConfigJson,
+		},
+	}
+	testMarketPrices := []pricestypes.MarketPrice{
+		constants.TestMarketPrices[0],
+		constants.TestMarketPrices[1],
+	}
+	testPerps := []perptypes.Perpetual{
+		constants.BtcUsd_0DefaultFunding_0AtomicResolution,
+		constants.EthUsd_0DefaultFunding_9AtomicResolution,
+	}
+	// TODO (TRA-118): store vault strategy constants in x/vault state.
+	minBaseSpreadPpm := uint32(3_000)                   // 30bps
+	baseSpreadMinPriceChangePremiumPpm := uint32(1_500) // 15bps
+	orderExpirationSeconds := uint32(5)                 // 5 seconds
+
+	// Initialize tApp and ctx with above test parameters.
+	tApp := testapp.NewTestAppBuilder(t).WithGenesisDocFn(func() (genesis types.GenesisDoc) {
+		genesis = testapp.DefaultGenesis()
+		// Initialize prices module with test markets.
+		testapp.UpdateGenesisDocWithAppStateForModule(
+			&genesis,
+			func(genesisState *pricestypes.GenesisState) {
+				genesisState.MarketParams = testMarketParams
+				genesisState.MarketPrices = testMarketPrices
+			},
+		)
+		// Initialize perpetuals module with test perpetuals.
+		testapp.UpdateGenesisDocWithAppStateForModule(
+			&genesis,
+			func(genesisState *perptypes.GenesisState) {
+				genesisState.LiquidityTiers = constants.LiquidityTiers
+				genesisState.Perpetuals = testPerps
+			},
+		)
+		// Initialize clob module with test clob pairs.
+		testapp.UpdateGenesisDocWithAppStateForModule(
+			&genesis,
+			func(genesisState *clobtypes.GenesisState) {
+				genesisState.ClobPairs = testClobPairs
+			},
+		)
+		return genesis
+	}).Build()
+	ctx := tApp.InitChain()
+
+	// Calculate subticks of BTC and ETH.
+	btcSubticks := clobtypes.PriceToSubticks(
+		testMarketPrices[0],
+		testClobPairs[0],
+		testPerps[0].Params.AtomicResolution,
+		lib.QuoteCurrencyAtomicResolution,
+	)
+	ethSubticks := clobtypes.PriceToSubticks(
+		testMarketPrices[1],
+		testClobPairs[1],
+		testPerps[1].Params.AtomicResolution,
+		lib.QuoteCurrencyAtomicResolution,
+	)
+	// Calculate spreads of BTC and ETH.
+	// - spread = max(minBaseSpreadPpm, baseSpreadMinPriceChangePremiumPpm + minPriceChangePpm)
+	// - btcSpreadPpm = max(3_000, 1_500+50) = 3_000
+	btcSpreadPpm := lib.Max(minBaseSpreadPpm, baseSpreadMinPriceChangePremiumPpm+testMarketParams[0].MinPriceChangePpm)
+	// - ethSpreadPpm = max(3_000, 1_500+4_200) = 5_700
+	ethSpreadPpm := lib.Max(minBaseSpreadPpm, baseSpreadMinPriceChangePremiumPpm+testMarketParams[1].MinPriceChangePpm)
+	// Calculate order good-till-block-time.
+	orderGtbt := &clobtypes.Order_GoodTilBlockTime{
+		GoodTilBlockTime: uint32(ctx.BlockTime().Unix()) + orderExpirationSeconds,
+	}
+
+	tests := map[string]struct {
+		/* --- Setup --- */
+		// Vault ID.
+		vaultId vaulttypes.VaultId
+
+		/* --- Expectations --- */
+		// Expected orders.
+		expectedOrders []clobtypes.Order
+	}{
+		"Get orders from Vault for Clob Pair 0": {
```

Comment:
nit: Non-blocking, the test cases are quite verbose and probably can be simplified to hold several parameters about each order rather than all the entire orders / constants could be declared for the orders.

Reply:
yeah agreed. let me simplify these tests